### DISCUSSION TITLE
No Acceptable Proposals

### DISCUSSION THREAD

**Discussion by aacklin** [strongswan/strongswan#979] [Q&A] [answered]
Hi. I'm trying to configure connection to Azure P2S from Ubuntu + it continues to fail. I've tried several of the resolutions from other posts with no success, even the post to change the leftid value to "%client.".  The snippet from the logs are below, Any assistance you can provide is greatly appreciated.

ipsec up azure
initiating IKE_SA azure[1] to 13.75.xx.xx
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from 10.10.xx.xx[500] to 13.75.xx.xx[500] (1128 bytes)
received packet: from 13.75.xx.xx[500] to 10.10.xx.xx[500] (38 bytes)
parsed IKE_SA_INIT response 0 [ N(INVAL_KE) ]
peer didn't accept DH group ECP_256, it requested ECP_384
initiating IKE_SA azure[1] to 13.75.xx.xx
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from 10.10.xx.xx[500] to 13.75.xx.xx[500] (1160 bytes)
received packet: from 13.75.xx.xx[500] to 10.10.xx.xx[500] (357 bytes)
parsed IKE_SA_INIT response 0 [ SA KE No N(FRAG_SUP) N(NATD_S_IP) N(NATD_D_IP) V V CERTREQ ]
received MS NT5 ISAKMPOAKLEY v9 vendor ID
received MS-Negotiation Discovery Capable vendor ID
selected proposal: IKE:AES_GCM_16_256/PRF_HMAC_SHA2_384/ECP_384
local host is behind NAT, sending keep alives
received cert request for "CN=P2SRootCert"
sending cert request for "C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root CA"
establishing CHILD_SA azure{1}
generating IKE_AUTH request 1 [ IDi N(INIT_CONTACT) CERTREQ CPRQ(ADDR DNS) SA TSi TSr N(MOBIKE_SUP) N(NO_ADD_ADDR) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
sending packet: from 10.10.xx.xx[4500] to 13.75.xx.xx[4500] (354 bytes)
received packet: from 13.75.xx.xx[4500] to 10.10.xx.xx[4500] (1276 bytes)
parsed IKE_AUTH response 1 [ EF(1/3) ]
received fragment #1 of 3, waiting for complete IKE message
received packet: from 13.75.xx.xx[4500] to 10.10.xx.xx[4500] (1276 bytes)
parsed IKE_AUTH response 1 [ EF(2/3) ]
received fragment #2 of 3, waiting for complete IKE message
received packet: from 13.75.xx.xx[4500] to 10.10.xx.xx[4500] (1244 bytes)
parsed IKE_AUTH response 1 [ EF(3/3) ]
received fragment #3 of 3, reassembled fragmented IKE message (3624 bytes)
parsed IKE_AUTH response 1 [ IDr CERT CERT AUTH EAP/REQ/ID ]
received end entity cert "C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN.cloudapp.net"
received issuer cert "C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA"
  using certificate "C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN.cloudapp.net"
  using untrusted intermediate certificate "C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA"
checking certificate status of "C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN.cloudapp.net"
  requesting ocsp status from 'http://ocsp.digicert.com' ...
  ocsp response correctly signed by "C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA"
  ocsp response is valid: until Apr 07 18:48:01 2022
certificate status is good
  using trusted ca certificate "C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root CA"
checking certificate status of "C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA"
ocsp response verification failed, no signer certificate '0f:80:61:1c:82:31:61:d5:2f:28:e7:8d:46:38:b4:2c:e1:c6:d9:e2' found
  requesting ocsp status from 'http://ocsp.digicert.com' ...
  ocsp response correctly signed by "C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root CA"
  ocsp response is valid: until Apr 07 06:13:52 2022
certificate status is good
certificate policy 2.23.140.1.2.2 for 'C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN.cloudapp.net' not allowed by trustchain, ignored
  reached self-signed root ca with a path length of 1
authentication of 'C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN.cloudapp.net' with RSA signature successful
server requested EAP_IDENTITY (id 0x00), sending 'P2SChildCert'
EAP_IDENTITY not supported, sending EAP_NAK
generating IKE_AUTH request 2 [ EAP/RES/NAK ]
sending packet: from 10.10.xx.xx[4500] to 13.75.xx.xx[4500] (67 bytes)
received packet: from 13.75.xx.xx[4500] to 10.10.xx.xx[4500] (72 bytes)
parsed IKE_AUTH response 2 [ N(MS_STATUS(1244)) ]
received MS_NOTIFY_STATUS notify error
establishing connection 'azure' failed

--------------------------------------------------------------------------------------------------------------------

/etc/ipsec.conf:

conn azure
      keyexchange=ikev2
      type=tunnel
      leftfirewall=yes
      left=%any
      leftauth=eap-tls
      leftid=P2SChildCert
      right=SomeAzureP2SVPN.cloudapp.net
      rightid=%SomeAzureP2SVPN.cloudapp.net
      rightsubnet=0.0.0.0/0
      leftsourceip=%config
      auto=add

**Comment by tobiasbrunner** [comment]
> server requested EAP_IDENTITY (id 0x00), sending 'P2SChildCert'
> EAP_IDENTITY not supported, sending EAP_NAK

You need the _eap-identity_ plugin (see [this FAQ entry](https://wiki.strongswan.org/projects/strongswan/wiki/FAQ#Plugin-is-missing)).

**Comment by aacklin** [comment]
Thanks @tobiasbrunner.  This resolved the EAP authentication issue but the connection continues to fail.  From the logs, there is a proposal mismatch but I'm unsure if this is the right path or something else is missing.  There is also an issue with setting the DNS but I can address that outside of this thread.  My main concern is establishing the VPN connection.

initiating IKE_SA azure[1] to 13.x.x.x
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from 10.10.x.x[500] to 13.x.x.x[500] (1128 bytes)
received packet: from 13.x.x.x[500] to 10.10.x.x[500] (38 bytes)
parsed IKE_SA_INIT response 0 [ N(INVAL_KE) ]
peer didn't accept DH group ECP_256, it requested ECP_384
initiating IKE_SA azure[1] to 13.x.x.x
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from 10.10.x.x[500] to 13.x.x.x[500] (1160 bytes)
received packet: from 13.x.x.x[500] to 10.10.x.x[500] (357 bytes)
parsed IKE_SA_INIT response 0 [ SA KE No N(FRAG_SUP) N(NATD_S_IP) N(NATD_D_IP) V V CERTREQ ]
received MS NT5 ISAKMPOAKLEY v9 vendor ID
received MS-Negotiation Discovery Capable vendor ID
selected proposal: IKE:AES_GCM_16_256/PRF_HMAC_SHA2_384/ECP_384
local host is behind NAT, sending keep alives
received cert request for "CN=P2SRootCert"
sending cert request for "C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root CA"
establishing CHILD_SA azure{1}
generating IKE_AUTH request 1 [ IDi N(INIT_CONTACT) CERTREQ CPRQ(ADDR DNS) SA TSi TSr N(MOBIKE_SUP) N(NO_ADD_ADDR) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
sending packet: from 10.10.x.x[4500] to 13.x.x.x[4500] (354 bytes)
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (1276 bytes)
parsed IKE_AUTH response 1 [ EF(1/3) ]
received fragment #1 of 3, waiting for complete IKE message
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (1276 bytes)
parsed IKE_AUTH response 1 [ EF(2/3) ]
received fragment #2 of 3, waiting for complete IKE message
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (1244 bytes)
parsed IKE_AUTH response 1 [ EF(3/3) ]
received fragment #3 of 3, reassembled fragmented IKE message (3624 bytes)
parsed IKE_AUTH response 1 [ IDr CERT CERT AUTH EAP/REQ/ID ]
received end entity cert "C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN"
received issuer cert "C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA"
  using certificate "C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN"
  using untrusted intermediate certificate "C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA"
checking certificate status of "C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN"
  requesting ocsp status from 'http://ocsp.digicert.com' ...
  ocsp response correctly signed by "C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA"
  ocsp response is valid: until Apr 25 18:48:01 2022
certificate status is good
  using trusted ca certificate "C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root CA"
checking certificate status of "C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA"
ocsp response verification failed, no signer certificate '0f:80:61:1c:82:31:61:d5:2f:28:e7:8d:46:38:b4:2c:e1:c6:d9:e2' found
  requesting ocsp status from 'http://ocsp.digicert.com' ...
  ocsp response correctly signed by "C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root CA"
  ocsp response is valid: until Apr 25 06:13:52 2022
certificate status is good
certificate policy 2.23.140.1.2.2 for 'C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN' not allowed by trustchain, ignored
  reached self-signed root ca with a path length of 1
authentication of 'C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN' with RSA signature successful
server requested EAP_IDENTITY (id 0x00), sending 'P2SChildCert'
generating IKE_AUTH request 2 [ EAP/RES/ID ]
sending packet: from 10.10.x.x[4500] to 13.x.x.x[4500] (78 bytes)
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (72 bytes)
parsed IKE_AUTH response 2 [ EAP/REQ/TLS ]
server requested EAP_TLS authentication (id 0x01)
generating IKE_AUTH request 3 [ EAP/RES/TLS ]
sending packet: from 10.10.x.x[4500] to 13.x.x.x[4500] (251 bytes)
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (1064 bytes)
parsed IKE_AUTH response 3 [ EAP/REQ/TLS ]
generating IKE_AUTH request 4 [ EAP/RES/TLS ]
sending packet: from 10.10.x.x[4500] to 13.x.x.x[4500] (67 bytes)
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (1064 bytes)
parsed IKE_AUTH response 4 [ EAP/REQ/TLS ]
generating IKE_AUTH request 5 [ EAP/RES/TLS ]
sending packet: from 10.10.x.x[4500] to 13.x.x.x[4500] (67 bytes)
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (1064 bytes)
parsed IKE_AUTH response 5 [ EAP/REQ/TLS ]
generating IKE_AUTH request 6 [ EAP/RES/TLS ]
sending packet: from 10.10.x.x[4500] to 13.x.x.x[4500] (67 bytes)
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (728 bytes)
parsed IKE_AUTH response 6 [ EAP/REQ/TLS ]
negotiated TLS 1.2 using suite TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
received TLS server certificate 'C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN'
received TLS intermediate certificate 'C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA'
  using certificate "C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN"
  using untrusted intermediate certificate "C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA"
checking certificate status of "C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN"
  ocsp response correctly signed by "C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA"
  ocsp response is valid: until Apr 25 18:48:01 2022
  using cached ocsp response
certificate status is good
  using trusted ca certificate "C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root CA"
checking certificate status of "C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA"
ocsp response verification failed, no signer certificate '0f:80:61:1c:82:31:61:d5:2f:28:e7:8d:46:38:b4:2c:e1:c6:d9:e2' found
  ocsp response correctly signed by "C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root CA"
  ocsp response is valid: until Apr 25 06:13:52 2022
  using cached ocsp response
certificate status is good
certificate policy 2.23.140.1.2.2 for 'C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN' not allowed by trustchain, ignored
  reached self-signed root ca with a path length of 1
sending TLS peer certificate 'CN=P2SChildCert'
generating IKE_AUTH request 7 [ EAP/RES/TLS ]
sending packet: from 10.10.x.x[4500] to 13.x.x.x[4500] (1085 bytes)
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (72 bytes)
parsed IKE_AUTH response 7 [ EAP/REQ/TLS ]
generating IKE_AUTH request 8 [ EAP/RES/TLS ]
sending packet: from 10.10.x.x[4500] to 13.x.x.x[4500] (313 bytes)
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (136 bytes)
parsed IKE_AUTH response 8 [ EAP/REQ/TLS ]
generating IKE_AUTH request 9 [ EAP/RES/TLS ]
sending packet: from 10.10.x.x[4500] to 13.x.x.x[4500] (67 bytes)
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (72 bytes)
parsed IKE_AUTH response 9 [ EAP/SUCC ]
EAP method EAP_TLS succeeded, MSK established
authentication of 'P2SChildCert' (myself) with EAP
generating IKE_AUTH request 10 [ AUTH ]
sending packet: from 10.10.x.x[4500] to 13.x.x.x[4500] (113 bytes)
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (328 bytes)
parsed IKE_AUTH response 10 [ AUTH CPRP(ADDR DNS) SA TSi TSr ]
authentication of 'C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN' with EAP successful
IKE_SA azure[1] established between 10.10.x.x[P2SChildCert]...13.x.x.x[C=US, ST=Washington, L=Redmond, O=Microsoft Corporation, CN=SomeAzureP2SVPN]
scheduling reauthentication in 10015s
maximum IKE_SA lifetime 10555s
**adding DNS server failed
adding DNS server failed
handling INTERNAL_IP4_DNS attribute failed**
installing new virtual IP PrivateIP
**received proposals: ESP:AES_GCM_16_256/NO_EXT_SEQ
configured proposals: ESP:AES_CBC_128/AES_CBC_192/AES_CBC_256/HMAC_SHA2_256_128/HMAC_SHA2_384_192/HMAC_SHA2_512_256/HMAC_SHA1_96/AES_XCBC_96/NO_EXT_SEQ, ESP:AES_GCM_16_128/AES_GCM_16_192/AES_GCM_16_256
no acceptable proposal found
failed to establish CHILD_SA, keeping IKE_SA
sending DELETE for ESP CHILD_SA with SPI c9790103**
generating INFORMATIONAL request 11 [ D ]
sending packet: from 10.10.x.x[4500] to 13.x.x.x[4500] (69 bytes)
received packet: from 13.x.x.x[4500] to 10.10.x.x[4500] (72 bytes)
parsed INFORMATIONAL response 11 [ D ]
establishing connection 'azure' failed

**Comment by aacklin** [comment, marked as answer]
Found the solution to this problem.
Added the following lines to ipsec.conf + it worked!

**ike=chacha20poly1305-sha512-curve25519-prfsha512,aes256gcm16-sha384-prfsha384-ecp384,aes256-sha1-modp1024,aes128-sha1-modp1024,3des-sha1-modp1024!
esp=chacha20poly1305-sha512,aes256gcm16-ecp384,aes256-sha256,aes256-sha1,3des-sha1!**

**Reply by yfChaw** [reply]
Thank you very much!

**Reply by ejohn20** [reply]
I spent hours troubleshooting this. Thank you so much for posting this! 🙏
